pvlib/celltemp.py

```diff
@@ -0,0 +1,183 @@
+}
+
+
+def sapm(poa_global, temp_air, wind_speed, a, b, deltaT, irrad_ref=1000):
+    r'''
+    Estimate cell and module temperatures per the Sandia PV Array
+    Performance Model [1].
+
+    Parameters
+    ----------
+    poa_global : float or Series
+        Total incident irradiance [W/m^2].
+
+    temp_air : float or Series
+        Ambient dry bulb temperature [C].
+
+    wind_speed : float or Series
+        Wind speed at a height of 10 meters [m/s].
+
+    a : float
+        Parameter :math:`a` in :eq:`sapm1`.
+
+    b : float
+        Parameter :math:`b` in :eq:`sapm1`.
+
+    deltaT : float
+        Parameter :math:`\Delta T` in :eq:`sapm2` [C].
+
+    irrad_ref : float, default 1000
+        Reference irradiance, parameter :math:`E_{0}` in :eq:`sapm2` [W/m^2].
+
+    Returns
+    --------
+    DataFrame with columns 'temp_cell' and 'temp_module'.
+    Values in degrees C.
+
+    Notes
+    -----
+    The model for cell temperature :math:`T_{C}` and module temperature
+    :math:`T_{m}` is given by a pair of equations (Eq. 11 and 12 in [1]).
+
+    .. math::
+       :label: sapm1
+
+        T_{m} = E \times \exp (a + b \times WS) + T_{a}
+
+    .. math::
+       :label: sapm2
+
+        T_{C} = T_{m} + \frac{E}{E_{0}} \Delta T
+
+    Inputs to the model are plane-of-array irradiance :math:`E` (W/m2) and
+    ambient air temperature :math:`T_{a}` (C). Model outputs are surface
+    temperature at the back of the module :math:`T_{m}` and cell temperature
+    :math:`T_{C}`. Model parameters depend both on the module construction and
+    its mounting. Parameter sets are provided in [1] for representative modules
+    and mounting.
+
+    +---------------+----------------+-------+---------+---------------------+
+    | Module        | Mounting       | a     | b       | :math:`\Delta T [C]`|
+    +===============+================+=======+=========+=====================+
+    | glass/glass   | open rack      | -3.47 | -0.0594 | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/glass   | close roof     | -2.98 | -0.0471 | 1                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | open rack      | -3.56 | -0.075  | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | insulated back | -2.81 | -0.0455 | 0                   |
+    +---------------+----------------+-------+---------+---------------------+
+
+    References
+    ----------
+    [1] King, D. et al, 2004, "Sandia Photovoltaic Array Performance
+    Model", SAND Report 3535, Sandia National Laboratories, Albuquerque,
+    NM.
+
+    '''
+    temp_module = pd.Series(poa_global * np.exp(a + b * wind_speed) + temp_air)
+    temp_cell = temp_module + (poa_global / irrad_ref) * (deltaT)
+    return pd.DataFrame({'temp_cell': temp_cell, 'temp_module': temp_module})
```

Comment:
I think getting a DataFrame back even for scalars is not very nice.  I suggest a separate function for the second equation, which could be useful in other contexts.  And the `sapm` function could have a boolean parameter to select either cell or module temperature as output.